includes/Services/InstaMigrateService.php

```diff
@@ -26,82 +19,55 @@ class InstaMigrateService {
 	private $insta_api_key = '';
 
 	/**
-	 * Retry count
+	 * Tracker class instance.
 	 *
-	 * @var int $count
+	 * @var Tracker $tracker
 	 */
-	private $count = 0;
+	private $tracker;
 
 	/**
 	 * Set required api keys for insta to initiate the migration
 	 */
 	public function __construct() {
-		$encrypt             = new Encryption();
-		$this->insta_api_key = $encrypt->decrypt( get_option( 'newfold_insta_api_key', false ) );
-		if ( ! $this->insta_api_key ) {
-			$this->insta_api_key = UtilityService::get_insta_api_key( BRAND_PLUGIN );
-			update_option( 'newfold_insta_api_key', $encrypt->encrypt( $this->insta_api_key ) );
-		}
+		$this->tracker = new Tracker();
+		$this->tracker->reset_track_file();
+		$instawp_get_key_step = new GetInstaWpApiKey( $this->tracker );
+		$instawp_get_key_step->set_status( 'running' );
```

Comment:
Let’s define some standardized status slugs in `AbstractStep`, something like:
```
$statuses = array(
    'running'   => 'running',
    'completed' => 'completed',
    'failed'    => 'failed',
    'aborted'   => 'aborted'
);
```
Then, we can use $statuses['running'] instead of hardcoding values everywhere. This even though redundant will make it easier to update statuses across the codebase if needed in the future.